manifests/repo.pp

```diff
@@ -1,148 +1,239 @@
+#
+# @param enabled
+#   enabled setting in the repository file.
+#   True enables, false disables this feature. When empty, it will not be present.
+#
+# @param enablegroups
+#   enablegroups setting in the repository file.
+#   True enables, false disables this feature. When empty, it will not be present.
+#
+# @param gpgcheck
+#   gpgcheck setting in the repository file.
+#   True enables, false disables this feature. When empty, it will not be present.
+#
+# @param keepalive
+#   keepalive setting in the repository file.
+#   True enables, false disables this feature. When empty, it will not be present.
+#
+# @param repo_gpgcheck
+#   repo_gpgcheck setting in the repository file.
+#   True enables, false disables this feature. When empty, it will not be present.
+#
+# @param skip_if_unavailable
+#   skip_if_unavailable setting in the repository file.
+#   True enables, false disables this feature. When empty, it will not be present.
+#
+# @param ssl_check_cert_permissions
+#   ssl_check_cert_permissions setting in the repository file.
+#   True enables, false disables this feature. When empty, it will not be present.
+#
+# @param sslverify
+#   sslverify setting in the repository file.
+#   True enables, false disables this feature. When empty, it will not be present.
+#
+# @param bandwidth
+#   bandwidth setting in the repository file.
+#   Takes any integer. When empty, it will not be present.
+#
+# @param cost
+#   cost setting in the repository file.
+#   Takes any integer. When empty, it will not be present.
+#
+# @param mirrorlist_expire
+#   mirrorlist_expire setting in the repository file.
+#   Takes any integer. When empty, it will not be present.
+#
+# @param retries
+#   retries setting in the repository file.
+#   Takes any integer. When empty, it will not be present.
+#
+# @param timeout
+#   timeout settingg in the repository file.
+#   Takes any integer. When empty, it will not be present.
+#
+# @param description
+#   name setting in the repository file.
+#   Defaults to the name of the defined type. When empty, it will not be present.
+#
+# @param password
+#   password setting in the repository file. When empty, it will not be present.
+#
+# @param proxy_password
+#   proxy_password setting in the repository file. When empty, it will not be present.
+#
+# @param proxy_username
+#   proxy_username setting in the repository file. When empty, it will not be present.
+#
+# @param repositoryid
+#   repositoryid setting in the repository file. When empty, it will not be present.
+#
+# @param username
+#   username setting in the repository file. When empty, it will not be present.
+#
+# @param baseurl
+#   baseurl setting in the repository file. Accepts HTTP/HTTPS URLs.
+#   When empty, it will not be present.
+#
+# @param gpgcakey
+#   gpgcakey setting in the repository file. Accepts HTTP/HTTPS URLs.
+#   When empty, it will not be present.
+#
+# @param gpgkey
+#   gpgkey setting in the repository file.  Accepts HTTP/HTTPS URLs.
```

Comment:
This should also accept `file://` URL's.

This should accept multiple lines. The format is like this


```
gpgkey=file:///key1
       file:///key2
```

Notice there is no equal on the other lines. Perhaps use

```ruby
.join("\n       ")
```

Reply:
fixed this also for baseurl which do also allow the same URLs in "Allow ftp and file URLs for baseurl and gpgkey".

For easier spec testing I have choosen to have two spaces in both cases:
```
.join("\n  ")
```

Reply:
Thanks! I was really surprised that the config file could be line space delimited instead of comma delimited for certain options. Very odd.